drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -214,10 +245,86 @@ local function set_setpoint(setpoint)
       value = f_to_c(value)
     end
 
+    -- Gather cached setpoint values when considering setpoint limits
+    -- Note cached values should always exists, but defaults are chosen just in case to prevent
+    -- nil operation errors, and deadband logic from triggering.
+    local cooling_setpoint = device.state_cache.main.thermostatCoolingSetpoint.coolingSetpoint or { value = 100 }
+    if cooling_setpoint and cooling_setpoint.unit == "F"then
+      cooling_setpoint = f_to_c(cooling_setpoint.value)
+    else
+      cooling_setpoint = cooling_setpoint.value
+    end
+    local heating_setpoint = device.state_cache.main.thermostatHeatingSetpoint.heatingSetpoint or { value = 0 }
```

Comment:
we prefer to use `device:get_latest_state` here rather than accessing the cache directly

the second return value has the unit you're looking for:

```
--- @return any, any The first return value is the state.value present for the attribute, the second return is the state
---                  table (e.g. it would include both the value and unit keys if both are present)
```